pd-recover: add `--data-dir` in `--force-new-cluster` (#22570)
pingcap/docs: pd-recover.md

### Step 2: Start the surviving PD node

Start the surviving PD node using the `--force-new-cluster` startup parameter, and ensure that the node uses the original data directory of the surviving node. You can specify it explicitly in the command line with `--data-dir`, or configure `data-dir` in `conf/pd.toml` in advance. For example:

```shell
./bin/pd-server --force-new-cluster --name=pd-127.0.0.10-2379 --data-dir=/path/to/existing/pd/data --client-urls=http://0.0.0.0:2379 --advertise-client-urls=http://127.0.0.1:2379 --peer-urls=http://0.0.0.0:2380 --advertise-peer-urls=http://127.0.0.1:2380 --config=conf/pd.toml
```

> > - If `--data-dir` is not specified in the command line, ensure that `data-dir` in `conf/pd.toml` correctly points to the original data directory of the surviving PD node, otherwise `pd-recover` may fail in subsequent operations.

Author: ```suggestion

> **Note:**
>
> - If `--data-dir` is not specified in the command line, ensure that `data-dir` in `conf/pd.toml` correctly points to the original data directory of the surviving PD node. Otherwise `pd-recover` might fail in subsequent operations.
```

### Step 2: Start the surviving PD node

> Start the surviving PD node using the `--force-new-cluster` startup parameter, and ensure that the node uses the original data directory of the surviving node. You can specify it explicitly in the command line with `--data-dir`, or configure `data-dir` in `conf/pd.toml` in advance. For example:

Author: ```suggestion
Start the surviving PD node using the `--force-new-cluster` startup parameter, and ensure that the node uses the original data directory. You can specify it explicitly in the command line with `--data-dir`, or configure `data-dir` in `conf/pd.toml` in advance. For example:
```

```shell
./bin/pd-server --force-new-cluster --name=pd-127.0.0.10-2379 --data-dir=/path/to/existing/pd/data --client-urls=http://0.0.0.0:2379 --advertise-client-urls=http://127.0.0.1:2379 --peer-urls=http://0.0.0.0:2380 --advertise-peer-urls=http://127.0.0.1:2380 --config=conf/pd.toml
```

> **Note:**
>
> - If `--data-dir` is not specified in the command line, ensure that `data-dir` in `conf/pd.toml` correctly points to the original data directory of the surviving PD node. Otherwise `pd-recover` might fail in subsequent operations.

> > - If the original data directory of the surviving node is specified in both `conf/pd.toml` and the `--data-dir` flag, the `conf/pd.toml` configuration takes precedence.

Reviewer: ```suggestion
> - - If `data-dir` is specified in both `conf/pd.toml` and the command-line arguments, the `data-dir` setting in `conf/pd.toml` takes precedence.
```

> > - If the original data directory of the surviving node is specified in both `conf/pd.toml` and the `--data-dir` flag, the `conf/pd.toml` configuration takes precedence.

Author: ```suggestion
> - If you specify the original data directory in both `conf/pd.toml` and `--data-dir` in the command line, `data-dir` in the `conf/pd.toml` configuration takes precedence.
```
